### ISSUE TITLE
Installation requirements for python 3.7

### ISSUE THREAD

**Issue by andreportela** [base4sistemas/pyescpos#17]
My environment:
**mac OS X 10.14.6
python 3.7
Epson TM-T20 installed on USB**

I tried to install it using `pip install PyESCPOS` but it failed with on PyBluez installation. It automatically tries to install PyBluez 0.23 but it says it's compatible with PyBluez 0.22.

I installed it using `pip install --no-deps PyESCPOS` it resumed the setup. 
I then tried to:
```python
from escpos.ifusb import USBConnection
from escpos.impl.epson import TMT20
```
it fails with: 
```python
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/__init__.py", line 28, in <module>
    from .conn import *
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/conn/__init__.py", line 24, in <module>
    from .bt import BluetoothConnection
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/conn/bt.py", line 25, in <module>
    import bluetooth
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
ModuleNotFoundError: No module named 'bluetooth'
```

I've tried to work around running a direct `pip install pybluez`. Then the exception stack trace changed to:
```python
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/__init__.py", line 28, in <module>
    from .conn import *
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/conn/__init__.py", line 24, in <module>
    from .bt import BluetoothConnection
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/conn/bt.py", line 27, in <module>
    from .. import config
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/work/virtual_environments/printer-api-8J7RRwy3/lib/python3.7/site-packages/escpos/config.py", line 25, in <module>
    from ConfigParser import SafeConfigParser
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
ModuleNotFoundError: No module named 'ConfigParser'
```

Now, even though I've successfully installed `pip install ConfigParser`. I get the same stack trace.

I can see that escpos/config.py:25 issues a 
```python
from ConfigParser import SafeConfigParser
```
whereas I suppose it should be 
```python
from configparser import SafeConfigParser
```
instead.

I have some questions about all this:
- Should the bluetooth (and other methods) support be strictly optional? I just need the printer usb driver;
- I don't really know, but I'm guessing this configparser import is meant to python2. Would it be ok to just do a conditional import there?
- Maybe #13 already addresses some stuff for this issue?

**Comment by fabianoengler** [comment]
@andreportela it seems all your issues are related to python3 support, which is addressed on my pr #13 , e.g. `ConfigParser` module was renamed on py3 and it should not be installed with pip, it is standard library.

Regarding your first point, I agree that optional stuff should have optional installation as well, but that is not how pyescpos is implemented currently, I guess there is some work going on that direction but not released yet. I actually thought about doing a PR about that too in the past, but since my last PR was completely ignored, I didn't felt much inclined to keep contributing, anyhow if you use my port for py3 you will see that although things like bluez are pulled automatically, they are inert code that are imported but not used, so no much harm done.

Try my branch and see if it works for you, it is a bit behind `upstream/master` (just a patch on top of 0.2) but it is working for me in production for almost a year now:
```
pip uninstall PyESCPOS
pip install git+https://github.com/fabianoengler/pyescpos.git
```

I'm not closely following upstream, but it seems they are going with py3 support, next release should work on py3 I believe but no idea when that's gonna be. You may try to install their master branch directly as well and see if it works.

**Comment by danielgoncalves** [comment]
Release 0.3 supports Python 3.6 to 3.8.

**Comment by MeanEYE** [comment]
> Release 0.3 supports Python 3.6 to 3.8.

No it doesn't. It fails with following:
```
Traceback (most recent call last):
  File "./label_maker.py", line 14, in <module>
    from escpos import BluetoothConnection
  File "/home/meaneye/Development/Workspace/Python/LabelMaker/lib/escpos/escpos/__init__.py", line 31, in <module>
    from .conn import *  # noqa: E402,F401,F403
  File "/home/meaneye/Development/Workspace/Python/LabelMaker/lib/escpos/escpos/conn/__init__.py", line 27, in <module>
    from .bt import BluetoothConnection
  File "/home/meaneye/Development/Workspace/Python/LabelMaker/lib/escpos/escpos/conn/bt.py", line 27, in <module>
    from future.utils import python_2_unicode_compatible
ModuleNotFoundError: No module named 'future'
```

**Comment by MeanEYE** [comment]
This is probably unrelated issue, which I will open but there are issues with Python 3.x. Don't have the time to fix them right now but will try at some point if someone else doesn't beat me to it.

**Comment by danielgoncalves** [comment]
Sorry for the lag, time is a valuable and scarce resource.

Optional dependencies was introduced in [this commit](https://github.com/base4sistemas/pyescpos/commit/58ebc1b544458803c4235f3fa80e8fa376b18ec2). These changes will be available at PyPI in the next 0.4 release, that (I hope) will be released soon.

In the mean time, clone the repository at master, make a source distribution and install from the tarball:

```
$ git clone https://github.com/base4sistemas/pyescpos.git
$ cd pyescpos
$ python -V
Python 3.7.7

$ python setup.py sdist
$ cd dist/
$ pip install PyESCPOS-0.3.tar.gz[usb]  # it will install with optional USB dependencies
```
